portfolio/src/main/webapp/js/comments.js

```diff
@@ -1,21 +1,39 @@
+/**
+ * Initializes the comment system.
+ *
+ * Adds a submission event listener to the comment form and loads the comments.
+ */
+function initCommentsSystem() {
+  const commentForm = document.getElementById("comment-form");
+  commentForm.addEventListener("submit", postComment);
+
+  loadComments();
+}
+
 /**
  * Gets comments from the '/comments' endpoint
  * and displays them in the comments section.
  */
 function loadComments() {
-  fetch('/comments')
-    .then(resp => resp.json())
-    .then(comments => {
-      console.log("Received comments: ");
-      console.log(comments);
-
-      removeCommentsOnPage();
-      comments.forEach((comment) => addCommentToPage(comment));  
-    })
-    .catch(err => {
-      console.error(err);
-      alert("Failed to populate comments.");
-    });
+  fetch('/comments').then(resp => {
+        if (resp.ok) {
+          return resp.json();
+        } else {
+          return resp.text().then(
+              text => Promise.reject(`Error ${resp.status}: ${text}`));
+        }
+      }).then(comments => {
+        console.log("Received comments: ");
+        console.log(comments);
+
+        removeCommentsOnPage();
+        comments.forEach((comment) => addCommentToPage(comment));  
+      }).catch(err => {
+        console.error(err);
+
+        addNotification(
+            "Failed to populate the comments section!", "alert-danger");
+      });
```

Comment:
nit: can we de-dent this block by 2 spaces?

Reply:
Okay I see now. They don't count as line breaks but block-like constructs and therefore are only spaced by 2. Will update.

Reply:
Yeah sorry I guess it's pretty arbitrary but I think that's how it works.

Reply:
It looked weird so I'm happy it's clear to me now!